projects/vdk-plugins/vdk-audit-hook/src/vdk/plugin/audit_hook/audit_hook_plugin.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+import sys
+from typing import List
+
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.api.plugin.plugin_registry import IPluginRegistry
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.internal.core.config import ConfigurationBuilder
+from vdk.plugin.audit_hook.audit_hook_config import add_definitions
+from vdk.plugin.audit_hook.audit_hook_config import AuditHookConfiguration
+
+
+class AuditHookPlugin:
+    @staticmethod
+    @hookimpl
+    def vdk_configure(config_builder: ConfigurationBuilder) -> None:
+        add_definitions(config_builder)
+
+    @hookimpl
+    def initialize_job(self, context: JobContext) -> None:
+        self._config = AuditHookConfiguration(context.core_context.configuration)
+
+        if not self._config.enabled():
+            return
+
+        not_permitted_events_list = self._config.not_permitted_events_list().split(";")
+
+        def _audit(event, args):
+            if any(
+                event in not_permitted_event
+                for not_permitted_event in not_permitted_events_list
+            ):
+                logging.getLogger(__name__).warning(
+                    f'[Audit Hook] Detected NOT permitted operation "{event}" with arguments "{args}"'
+                )
+
+                if self._config.exit_on_not_permitted_event():
+                    logging.getLogger(__name__).error(
+                        f'[Audit Hook] Terminating the data job due to the NOT permitted operation "{event}" with arguments "{args}"'
+                    )
+                    os._exit(0)
```

Comment:
This is a hard exit. .. I imagine it's because we do not want users to catch Exception and swallow it. 
 
 We should document it that if exit_on_not_permitted_event it's fully terminated - no clean up ,no per attempt notifications, etc.
 
 But this seems an error scenario. Shouln't this fail with non zero status?

Reply:
> This is a hard exit. .. I imagine it's because we do not want users to catch Exception and swallow it.
> 
> We should document it that if exit_on_not_permitted_event it's fully terminated - no clean up ,no per attempt notifications, etc.
> 

You are absolutely correct about that.

> But this seems an error scenario. Shouln't this fail with non zero status?

In my opinion, we should not retry the job in case of not permitted operation. As far as I remember If I use os._exit(1) (or something) the job will be retried, right? I remember that there was a discussion around that.

Reply:
Yes, but we will change that in the future. Especially with [Pod failure policy](https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-failure-policy) being introduced in Kubernetes, that would be much easier. 

 I'd prefer to at least make the exit status configurable . 

We can leave the default to zero. But we should explain why it's set to zero (so the job is not re-tried)

Reply:
Ok, I will change it.